.infrastructure/daemonset.yml

```
@@ -0,0 +1,38 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: daemonset
+  namespace: mateapp
+  labels:
+    app: daemonset
+spec:
+  selector:
+    matchLabels:
+      app: daemonset
+  template:
+    metadata:
+      labels:
+        app: daemonset
+    spec:
+      containers:
+        - name: busybox
+          image: busyboxplus:curl
+          resources:
+            requests:
+              cpu: "250m"
+              memory: "256Mi"
+            limits:
+              cpu: "500m"
+              memory: "512Mi"
+          ports:
+            - containerPort: 80
+              hostPort: 30008
+          command: [ "/bin/sh", "-c" ]
+          args:
+              - >
+                while true;
+                do
+                  echo "$(date): Checking todoapp...";
+                  curl -s http://todoapp:80 || echo "Failed";
```

The curl command currently masks failures: `curl -s http://todoapp:80 || echo "Failed"`. This pattern makes the overall command succeed even if curl fails (because `echo` returns 0), hiding failures from job status and logs. Use a failure-preserving, non-interactive form such as `curl -fsS http://todoapp:80` (and remove the `|| echo "Failed"`) so connection errors or non-2xx responses return non-zero and are observable in logs. This is required by the checklist to ensure failures are recorded and visible.